src/charm.py

```diff
@@ -163,63 +134,85 @@ def _on_install(self, _: InstallEvent) -> None:
         self.unit.status = MaintenanceStatus("Building srsLTE")
         shell(SRS_ENB_UE_BUILD_COMMAND)
 
-        self.unit.status = MaintenanceStatus("Generating configuration files")
+        self.unit.status = MaintenanceStatus("Copying example configuration files")
         copy_files(origin=CONFIG_ORIGIN_PATHS, destination=CONFIG_PATHS)
 
-        self.unit.status = MaintenanceStatus("Generating systemd files")
+        self.unit.status = MaintenanceStatus("Configuring srs env service")
         self._configure_srsenb_service()
-        self._configure_srsue_service()
 
         service_enable(SRS_ENB_SERVICE)
-        self._stored.installed = True
+        self._set_peer_data("installed", True)
 
     def _on_start(self, _: StartEvent) -> None:
         """Triggered on start event."""
+        if not self.unit.is_leader():
+            return
         self.unit.status = MaintenanceStatus("Starting srsenb")
         service_start(SRS_ENB_SERVICE)
-        self._stored.started = True
-        self.unit.status = self._get_current_status()
+        self._set_peer_data("started", True)
+        self.unit.status = ActiveStatus(self._active_status_msg)
 
     def _on_stop(self, _: StopEvent) -> None:
         """Triggered on stop event."""
+        if not self.unit.is_leader():
+            return
         self._reset_environment()
         service_stop(SRS_ENB_SERVICE)
-        self._stored.started = False
-        self.unit.status = self._get_current_status()
+        self._set_peer_data("started", False)
```

Comment:
Same here, shouldn't we check for the peer relation to be joined?

Reply:
Removed `peer_data` here